utilities/transactions/transaction_test.cc

```diff
@@ -473,14 +500,102 @@ TEST_P(TransactionTest, DeadlockCycleShared) {
   for (auto& t : threads) {
     t.join();
   }
+
+  auto dlock_buffer_before_resize = db->GetDeadlockInfoBuffer();
```

Comment:
Since we're testing something completely differently different now, I think we should either add a new test, or at least add a comment denoting this.

For a new test, you'll need to recreate the deadlocks, but you're doing that anyway below.

Some comments on what we're testing in each case would be helpful.